src/passes/OptimizeInstructions.cpp

```diff
@@ -1312,6 +1312,8 @@ struct OptimizeInstructions
                    !EffectAnalyzer(getPassOptions(), binary->left)
                       .hasSideEffects()) {
           return binary->right;
+        } else if (binary->op == EqInt64) {
+          return Builder(*getModule()).makeUnary(EqZInt64, binary->left);
```

Comment:
Where is this optimization done for `i32.eqz`? I would expect this optimization to be organized to be symmetrically located with the `i32` version.

Reply:
The other one appears to be here: https://github.com/WebAssembly/binaryen/blob/7fa8c78dc02501ba4cd6088b3664a1eb8b794cda/src/passes/OptimizeInstructions.cpp#L373 also handling various sign extension cases, which made me shy away from a quick fix as it seems there's more to it.

Reply:
Yeah, this is not structured in a very symmetrical way atm. It grew organically + some thoughts about checking common patterns early. So i32.eqz is much more common than i64.eqz, and hence they are differently located. It's very possible that's not worth it, and this should be refactored.

Reply:
OK, LGTM for now.